frontend/src/components/NavBar.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Divider,
+  createIcon,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverHeader,
+  PopoverBody,
+  PopoverArrow,
+  PopoverCloseButton,
+  Portal,
+  Link,
+} from "@chakra-ui/react";
+import { useRouter } from "next/router";
+
+// button abstraction to include the Link and styling
+const StyleButton = ({ children, link }: { children: any; link: string }) => {
+  const router = useRouter();
+  const isActive = router.pathname === link;
+  return (
+    <>
+      <Link href={link} width="100%">
+        <Button
+          justifyContent="flex-start"
+          bg="transparent"
+          height="9"
+          width="100%"
+          borderRadius={10}
+          // temporary until I figure out custom color schemes
+          style={{
+            background: isActive ? "#FAFA78" : "white",
+          }}
+          _hover={{
+            background: "#FAFA78 !important",
+          }}
+          isActive={isActive}
+        >
+          {children}
+        </Button>
+      </Link>
+    </>
+  );
+};
+
+const OverViewIcon = createIcon({
```

Comment:
Maybe we can move all these icons to a separate file? Then just import them as needed. I think that would make this page a bit more readable

Reply:
Just pushed the changes!